io/io/src/TGenCollectionProxy.cxx

```diff
@@ -958,6 +958,12 @@ TGenCollectionProxy *TGenCollectionProxy::InitializeEx(Bool_t silent)
                }
                break;
          }
+         if (!(fProperties & kIsEmulated) && newfValue->fType && !newfValue->fType->IsSyntheticPair()) {
+            if (!newfValue->fType->IsLoaded() && !newfValue->fType->HasInterpreterInfo())
```

Comment:
> IsLoaded() && !HasInterpreterInfo() then that's fine?

That state is 'impossible' (more exactly it would means that the dict.cxx was loaded by both the rootcpm and header file failed to load or be parsed).

> I guess my question is, how can you have HasInterpreterInfo() without IsLoaded()?

`IsLoaded()` means that there a dictionary file generated by `rootcling/genreflex` (thus have accelerator functions and pre-digested information).   `HasInterpreterInfo()` on the other hand means that Cling knows (or can know) about the class, for example if the header file was loaded on the prompt or if the class information is loaded as part of something else but does not have its own dictionary (for example there is a dictionary for `std::map<a,b>` but not for `pair<a,b>` and the header for `map, a and b` are loaded)

Reply:
> IsLoaded() && !HasInterpreterInfo() then that's fine?

This can happen (albeit it is a 'degredated' situation) when there is a dict.cxx file generated by `rootcling/genreflex` but the full interpreter information can not be loaded (for example was able to load the rootpcm files with the `TProtoClass` but was not able to load/parse the header file).

Reply:
> !newfValue->fType->IsLoaded() && !newfValue->fType->HasInterpreterInfo()

I think this is an expressive version of the potential combination (eg. I could probably use `HasDataMemberInfo()`) of testing as it clearly expresses that we need the TClass to have gotten its information either indirectly from `rootcling` (`IsLoaded()`) or directly from `Cling` (`HasInterpreterInfo()`).